### DISCUSSION TITLE
Changing world difficulty of Tmodloader changes world version also

### DISCUSSION THREAD

**Discussion by Lankoss** [Pryaxis/TShock#2532] [Q&A] [answered]
I'm trying to change my Calamity world on Tmodloader from Normal to Expert, it works no problem but when I load Tmodloader back up to play it, my world now has the (1.4 worlds not supported) message. The way I have been converting it is to cut/paste my Tmod world into the Terraria world folder (so that it shows up as an option within Tshock), editing the world mode then cut/paste it back into the Tmod world folder.

Should I be doing it a different way? Or maybe I should be using an older version of Tshock that doesn't automatically convert the world to 1.4 when changing the World difficulty?

Let me know, thanks :)

**Comment by Arthri** [comment, marked as answer]
Are you using normal tModLoader without 1.4 support? If so then it means TShock is loading the world into 1.4 and saving it that way causing it to become a 1.4 world.

A better option would to be use [TEdit](https://github.com/TEdit/Terraria-Map-Editor/releases/tag/3.10.18105.218) instead

**Reply by Lankoss** [reply]
Yes I am using the normal Tmod without 1.4 for a better playing experience. I realise Tshock is changing the file to 1.4, what I was wondering is, is there a way to change the world difficulty without changing the world version? I was thinking of maybe trying and older version of Tshock prior to 1.4 release. I will check out Tedit also.

**Reply by Arthri** [reply]
using an older version of TShock also works :+1:

**Reply by Lankoss** [reply]
Do you happen to know the latest version of Tshock that is compatible with the current NORMAL Tmodloader release? I was planning on just using the last release of Tshock prior to 1.4 support. Edit: I'm looking into TEdit now also :)

**Reply by Arthri** [reply]
iirc it's https://github.com/Pryaxis/TShock/releases/tag/v4.3.26. The TEdit link in my original post is also the latest non-1.4 version I could find

**Reply by Lankoss** [reply]
In the interim I did try changing it with the latest version of Tedit, and although it changed it to Expert (and there was no 1.4 error message this time), the world errored out on the loading screen and failed to load. I will try the versions you suggested :) Thanks for your help thus far.

**Reply by Lankoss** [reply]
> iirc it's https://github.com/Pryaxis/TShock/releases/tag/v4.3.26. The TEdit link in my original post is also the latest non-1.4 version I could find

I was able to edit the world into Expert using the version of TEdit you linked, everything seems to be working fine. Thanks for your help :)
